specification/purview/data-plane/Azure.Analytics.Purview.Scanning/stable/2023-09-01/scanningService.json

```diff
@@ -0,0 +1,9300 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/Scan"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Scanning Service.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Scans_Get": {
+            "$ref": "./examples/Scans_Get.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Scan"
+        ],
+        "description": "Deletes the scan associated with the data source.",
+        "operationId": "Scans_Delete",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/dataSourceName"
+          },
+          {
+            "$ref": "#/parameters/scanName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "Success."
+          },
+          "default": {
+            "description": "An error response received from the Scanning Service.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "String error code indicating what went wrong."
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Scans_Delete": {
+            "$ref": "./examples/Scans_Delete.json"
+          }
+        }
+      }
+    },
+    "/datasources/{dataSourceName}/scans": {
+      "get": {
+        "tags": [
+          "Scan"
+        ],
+        "description": "Lists scans in data source.",
+        "operationId": "Scans_ListByDataSource",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/dataSourceName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/ScanList"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Scanning Service.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Scans_ListByDataSource": {
+            "$ref": "./examples/Scans_ListByDataSource.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/datasources/{dataSourceName}/scans/{scanName}/runs/{runId}": {
+      "put": {
```

Comment:
Yes, the put method here seems not correct. I removed it and please refer to /datasources/{dataSourceName}/scans/{scanName}:run instead

Reply:
Hi Mike, I just discussed Shawn on this issue. And since we consider scan result as a sub-resource of the scan, and while many of the scan results are trigger-created. So we would like to keep this URL and still use PUT to ensure that this API is idempotent here.

Reply:
@mikekistler What I can see post won't work is that, scan run is a heavy operation and we want to make the API is idempotent, then common network retry won't result in unexpected additional scan runs for customer. In the other side, scan trigger with scheduler will create scan runs automatically, so for each scan, we have GET API "/datasources/{dataSourceName}/scans/{scanName}/runs" to return all scan runs for each scan, just like scan runs are the sub resources.  So if possible, we want to keep as it is for now, thanks.

Reply:
If you want to leave this as a PUT, then you need to use the LRO Create pattern in our guidelines, which means that the operation returns a 201 and there is a separate status monitor created to track completion.

But I don't really think that's what you want. If this is a true sub-resource, why is there no "delete" operation?

Perhaps you can come to API Stewardship office hours on Weds to discuss.

Reply:
@mikekistler  what's the guideline for idempotent action in Azure API? Btw, could you please share the detailed guideline as well, we can have a look at it and compare the possible options, thx.

Reply:
See https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#http-post-must-be-idempotent

Reply:
And the guideline for LRO PUT (Create) is:

https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#put-operation-with-additional-long-running-processing

Reply:
@mikekistler @weidongxu-microsoft  Thanks both for the quick review always. I offline had a discussion with @weidongxu-microsoft on this API, for the API itself, it intended to create a run and suppose to be done very quickly, but the underlying run resource can last up to 30 days, and the run resource will be kept for customers for at least 3 months, in this case, we understand it can be taken as sub resource of the scan since the action is used for operations that can be done in a short period, feel free to let us know if any more questions, thx.

Reply:
From Shawn's description, this "run" apears to be the VLRO (very long running operation) pattern we've been discussing with Jeffrey, as the underlying operation could be running for days or weeks to completion.

office hour https://aka.ms/azapi/officehours

Reply:
Maybe, but we haven't finalized that pattern yet, so I don't think we should be baking it into a GA API version until we know what it really looks like.

Reply:
@mikekistler We have some big customer waiting for GA version, they don't use the service unless it's GAed, what's the ETA to finalize that pattern? Can we leave it as is for 1st version and then apply the pattern in future versions when the pattern finalized?

Reply:
Once the API is GA we should not make changes that could break customers, and it's unknown whether the final VLRO pattern will be breaking from your current API. So I don't think we should GA the API as is.

If you need to GA, then I recommend you adopt the LRO pattern as currently defined in the Azure REST API Guidelines.

Reply:
Based on the discussion, I have changed the URL to /datasources/{dataSourceName}/scans/{scanName}:run, POST method